### PR TITLE
fix 14144

### LINKED ISSUE #14144: last_insert_id not returned when using update
### Which version of ShardingSphere did you use?
master branch

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
proxy

### Expected behavior
update id_gen set max_id = last_insert_id(max_id + 1) where type = 1;
the last_insert_id should be returned back to user

### Actual behavior
can't get last_insert_id like insert statement

### Reason analyze (If you can)
isReturnGeneratedKeys is only true on insert statement

### PR DESCRIPTION
**Pull request by roodkcab**
Fixes #14144.

Changes proposed in this pull request:
- get last_insert_id on update set it

### CODE REVIEW

**Comment by menghaoranss** [comment]
@roodkcab ci build failed.

**Comment by roodkcab** [comment]
@menghaoranss hi, is there anyway to trigger ci run again without push?

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/ProxySQLExecutor.java [hunk 1]
```diff
@@ -112,7 +115,16 @@ private boolean isExecuteDDLInPostgreSQLOpenGaussTransaction(final SQLStatement
         String schemaName = backendConnection.getConnectionSession().getSchemaName();
         Collection<ShardingSphereRule> rules = ProxyContext.getInstance().getContextManager().getMetaDataContexts().getMetaData(schemaName).getRuleMetaData().getRules();
         int maxConnectionsSizePerQuery = ProxyContext.getInstance().getContextManager().getMetaDataContexts().getProps().<Integer>getValue(ConfigurationPropertyKey.MAX_CONNECTIONS_SIZE_PER_QUERY);
-        boolean isReturnGeneratedKeys = executionContext.getSqlStatementContext().getSqlStatement() instanceof MySQLInsertStatement;
+        final SQLStatement stmt = executionContext.getSqlStatementContext().getSqlStatement();
+        boolean isReturnGeneratedKeys = stmt instanceof MySQLInsertStatement;
+        if (stmt instanceof MySQLUpdateStatement) {
+            for (final AssignmentSegment segment : ((MySQLUpdateStatement) stmt).getSetAssignment().getAssignments()) {
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/ProxySQLExecutor.java:121]: The value of `isReturnGeneratedKeys` controls whether we get generated key from ResultSet. Maybe looking for function `last_insert_id` in statement is not right.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

**Comment by terrymanu** [comment]
Hello, any update?

**Comment by TeslaCN** [comment]
Close due to no response.

**Pull request closed without merging**